### PR TITLE
Implemented the subject list page

### PR DESCRIPTION
**Pull request by once-upon-a-tam**
Until the [sign in pull request](https://github.com/Lern-PFR/lern-web/pull/32) as been merged, you'll need to replace the `getSubjects` selector according to the inline comments (`src/redux/selectors/subjects.js`)

### CODE REVIEW

File: src/i18n/en.json [hunk 1]
```diff
@@ -114,12 +114,27 @@
...
+				"hero": "Subjects list",
+				"sections": {
+					"active_subjects": "My currently active subjects",
+					"my_subjects": "Subjects of which I'm the author",
+					"other_available_subjects": "Other available subjects"
+				},
+				"search_input": {
+					"placeholder": "Search a subject by its name or author."
+				},
+				"no_data": "No subject found with the specified criterias."
```
Comment (frenchykiller) [inline src/i18n/en.json]: criteria is uncountable (no s)

File: src/i18n/en.json [hunk 2]
```diff
@@ -114,12 +114,27 @@
...
+			},
 			"links": {
 				"back_to_list": "Back to subject list",
 				"module": "Access module"
 			},
 			"details": {
-				"author_section": "{{author.firstName}} {{author.lastName}} - last update : {{lastUpdate}}"
+				"author_section": "{{author.firstname}} {{author.lastname}} - last update : {{lastUpdate, L - LT }}",
+				"modules": {
+					"no_data": "This subject doesn't have any module, yet."
```
Comment (frenchykiller) [inline src/i18n/en.json]: modules*

File: src/pages/subjects/SubjectListPage.jsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+			</StyledDiv>
+
+			<StyledDiv as="section">
+				<SubjectListSectionTitle>{t('subjects.list_page.sections.other_available_subjects')}</SubjectListSectionTitle>
+				<StyledList {...sectionList}>
+					{subjects && subjects.length > 0 && subjects.map((subject) => (<SubjectCard key={subject.id} {...subject} />))}
+					{(!subjects || subjects.length === 0) && <BodyCopy>{t('subjects.list_page.no_data')}</BodyCopy>}
+				</StyledList>
+			</StyledDiv>
+		</StyledDiv>
```
Comment (frenchykiller) [inline src/pages/subjects/SubjectListPage.jsx:83]: All 3 sections contain the same list of subjects (every existing subject)

File: src/pages/subjects/SubjectDetailsPage.jsx [hunk 4]
```diff
@@ -25,93 +26,57 @@ import { backToParentButtonContentLayout } from 'theme/buttonStyles';
...
-				<ModuleList modules={modules} />
-			</StyledDiv>
+			{!subject && <></>}
+			{subject && (
+				<>
+					<StyledDiv>
+						<StyledDiv {...subjectDetails}>
+							<Canon {...subjectName} tag="h1">{subject?.title}</Canon>
+							<Brevier {...subjectAuthor}>
+								{t('subjects.details.author_section', { author: subject?.author, lastUpdate: subject?.updatedAt })}
```
Comment (frenchykiller) [inline src/pages/subjects/SubjectDetailsPage.jsx:53]: Displays correct author name when author is viewing their subject, displays no name when another user views the subject

File: src/pages/subjects/SubjectListPage.jsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+	 * @description Handles the onChange event for the search input.
+	 *
+	 * @author Timothée Simon-Franza
+	 *
+	 * @param {object} e				The onChange event.
+	 * @param {object} e.target			The input triggering the event.
+	 * @param {string} e.target.value	The current value of the input.
+	 */
+	const handleSearchInputChange = useCallback(({ target: { value } }) => {
+		setSearchValue(value);
```
Comment (frenchykiller) [inline src/pages/subjects/SubjectListPage.jsx:45]: Does not support searching by author

**Review by frenchykiller — CHANGES_REQUESTED** [review]

**Comment by once-upon-a-tam** [comment]
[As a user, I want a way to see all available subjects.](https://trello.com/c/iLjAl1ku/62-as-a-user-i-want-a-way-to-see-all-available-subjects)

**Review by frenchykiller — APPROVED** [review]

**Pull request merged**